tests/test_init.py

```diff
@@ -282,3 +282,23 @@ def test_raise_invalid_threadcount():
         vmecpp.run(vmec_input, max_threads=-1)
     with pytest.raises(RuntimeError):
         vmecpp.run(vmec_input, max_threads=0)
+
+
+def test_VmecInput_validation():
+    test_file = TEST_DATA_DIR / "solovev.json"
+    vmec_input = vmecpp.VmecInput.from_file(test_file)
+
+    # Why do we not compare `json_dict = json.loads(test_file.read_text())` ?
+    # The test_file json may exclude fields that have default values,
+    # while the parsed versions should have all fields populated.
+    pywrapper_str = json.loads(vmec_input._to_cpp_vmecindatapywrapper().to_json())
```

Comment:
maybe clearer (`_str` is confusing, this is not a string?):
```suggestion
    pywrapper_dict_from_json = json.loads(vmec_input._to_cpp_vmecindatapywrapper().to_json())
```